src/app/MainPage.tsx

```diff
@@ -2,14 +2,22 @@
 import { useAtom } from "jotai";
 
 import { messageAtom } from "@/store/messageAtom.tsx";
+import Modal from "@/component/common/modal/Modal";
+import useModal from "@/hooks/useModal";
 
 function MainPage() {
   const [message] = useAtom(messageAtom);
+  const { open, close, isOpen } = useModal();
   return (
-    <div>
-      <span>welcome to layer 🎇</span>
-      <div>{message}</div>
-    </div>
+    <>
+      <div>
+        <span>welcome to layer 🎇</span>
+        <div onClick={open}>{message}</div>
+      </div>
+      <Modal onClose={close} isModalOpen={isOpen}>
+        <div>냠냠</div>
+      </Modal>
```

Comment:
```suggestion
      {isOpen &&
        <Modal onClose={close}>
          <div>냠냠</div>
        </Modal>
      }
```
요렇게 하고 Modal 안에서 스크롤 방지하는 로직을 useModal 훅으로 옮기면 isModalOpen props가 없어져도 될 것 같아요!

Reply:
prop로 넘기는 것보다 더 깔끔한 것 같아요~!! 감사합니다:)

Reply:
isOpen을 계속 써야하는 번거로움이 있을 것 같아서 코드를 개선해봤습니다~!!